.claude/hooks/format-and-lint.sh

```
@@ -0,0 +1,28 @@
+
+# Only process TypeScript files
+echo "$FILE_PATH" | grep -qE '\.(ts|tsx)$' || exit 0
+
+# Run prettier first
+npx prettier --write "$FILE_PATH" 2>/dev/null || true
+
+# Determine package and run appropriate linter
+PROJECT_DIR="${CLAUDE_PROJECT_DIR:-$(git rev-parse --show-toplevel 2>/dev/null)}"
+[ -z "$PROJECT_DIR" ] && exit 0
+
+REL_PATH="${FILE_PATH#$PROJECT_DIR/}"
```

Shellcheck (0.11.0)

[info] Expansions inside ${..} need to be quoted separately, otherwise they match as patterns.

(SC2295)